main/util/src/mill/util/CoursierSupport.scala

```diff
@@ -104,7 +108,7 @@ trait CoursierSupport {
             Agg.from(
               res.files
                 .map(os.Path(_))
-                .filter(path => path.ext == "jar" && resolveFilter(path))
+                .filter(resolveFilter)
```

Comment:
To preserve previous behavior, you should change the `resolveFilter` default to contain the `path.ext == "jar"` match, I guess.

Reply:
Answered in https://github.com/com-lihaoyi/mill/pull/3701#discussion_r1794821092